src/Microsoft.MobileBlazorBindings.WPF/BlazorWebView.cs

```diff
@@ -0,0 +1,531 @@
+            capturedRender.RootRenderHandle.Render(fragment);
+            //await capturedRender.Dispatcher.InvokeAsync(() => capturedRender.RootRenderHandle.Render(fragment ?? EmptyRenderFragment)).ConfigureAwait(false);
+        }
+
+        public async Task DispatchEvent(WebEventDescriptor eventDescriptor, string eventArgsJson)
+        {
+            if (eventDescriptor is null)
+            {
+                throw new ArgumentNullException(nameof(eventDescriptor));
+            }
+            if (_blazorHybridRenderer == null)
+            {
+                // we were shut down, but we got a render completion from javascript afterwards.
+                return;
+            }
+
+            await _blazorHybridRenderer.DispatchEventAsync(eventDescriptor, eventArgsJson).ConfigureAwait(false);
+        }
+
+#pragma warning disable CA1054 // Uri parameters should not be strings
+        public void NotifyLocationChanged(string uri, bool isInterceptedLink)
+#pragma warning restore CA1054 // Uri parameters should not be strings
+        {
+            _navigationManager.SetLocation(uri, isInterceptedLink);
+        }
+
+        public async Task OnRenderCompleted(long renderId, string errorMessageOrNull)
+        {
+            if (_blazorHybridRenderer == null)
+            {
+                // we were shut down, but we got a render completion from javascript afterwards.
+                return;
+            }
+            await _blazorHybridRenderer.OnRenderCompletedAsync(renderId, errorMessageOrNull).ConfigureAwait(false);
+        }
+
+        private static string BlazorAppScheme
+        {
+            get
+            {
+                // On Windows and Tizen, we can't use a custom scheme to host the initial HTML,
+                // because webview2 won't let you do top-level navigation to such a URL.
+                // On Linux/Mac, we must use a custom scheme, because their webviews
+                // don't have a way to intercept http:// scheme requests.
+                return "http";
+            }
+        }
+
+        /// <summary>
+        /// Gets the content type for the url.
+        /// </summary>
+        /// <param name="url">The url to use.</param>
+        /// <returns>The content type.</returns>
+        private static string GetContentType(string url)
+        {
+            if (FileExtensionContentTypeProvider.TryGetContentType(url, out var result))
+            {
+                return result;
+            }
+
+            return "application/octet-stream";
+        }
+
+        private static Stream SupplyFrameworkFile(string uri)
+        {
+            return uri switch
+            {
+                "framework://blazor.desktop.js" => BlazorAssets.GetBlazorDesktopJS(),
+                _ => throw new ArgumentException($"Unknown framework file: {uri}"),
+            };
+        }
+
+        //public void Dispose()
```

Comment:
Do you need that commented code?

Reply:
That's a good question. As far as I could tell, there isn't a standard disposal pattern for WPF controls, so I wasn't sure of the best pattern for this. I'm leaving the code there for future consideration. I'll add a comment for Future Us to have some idea.